platform/saivpp/vpplib/SwitchStateBaseAcl.cpp

```diff
@@ -563,6 +825,30 @@ sai_status_t SwitchStateBase::AclTblConfig(
     ordered_aces.sort(cmp_priority);
     // SWSS_LOG_NOTICE("# aces %u after sort", ordered_aces.size());
 
+    size_t n_entries = 0;
+    size_t n_tunterm_entries  = 0;
+    for (auto ace: ordered_aces) {
```

Comment:
This function is too long. We'd better break it into a few smaller functions. For example, for the part that converting ACE to acl_tbl_entries_t and sort can be one function. From sorted  acl_tbl_entries_t to vpp_acl_t and vpp_tunerm_acl_t can be another function.

Reply:
Conceptually with tunterm feature, each ACL table can have 1 or 2 ACL objects in VPP: one for regular ACL and one for tunterm ACL. In the create/update request, we go through the ACEs and convert them to 2 lists of vpp_acl_t and vpp_tunterm_acl_t (you can create a super data structure including the 2 lists). Then we pass the 2 lists along with table_id to a function. The function reads acl_swindex and tunterm_acl_swindex from table_id and does necessary operation to create/update/delete acl or tunterm acl from vpp.